tests/unit/common/automata/generated.cpp

```diff
@@ -250,3 +250,11 @@ TEST_CASE("No print after upload") {
     ex.consume("GET / HTTP/1.1\r\nPrint-After-Upload: false\r\n\r\n");
     REQUIRE_FALSE(ex.contains_enter(Names::PrintAfterUpload));
 }
+
+TEST_CASE("Digest auth whole") {
+    using test::http::Names;
+    TestExecution ex(http_request);
+    ex.consume("GET /api/version HTTP/1.1\r\nAuthorization: Digest username=\"user\", realm=\"Printer API\", nonce=\"dcd98b7102dd2f0e\", uri=\"/api/version\", response=\"684d849df474f295771de997e7412ea4\"\r\n\r\n");
```

Comment:
Can the uri here and in the first HTTP line differ?

Reply:
Not in a valid http request, only through some error or malicious stuff going on. But as of now we ignore the `uri` parameter in the `Authorization` header to save space, otherwise we would need to store another url and to be sure it fits allocate char array of size `MAX_URL_LEN` which is 100, so that's probably not worth it IMO.